### ISSUE TITLE
Remove Build Files from the Repository

### ISSUE THREAD

**Issue by alireza-keshavarz-d** [tamtaasatiani/tahmlib#8]
Hi! Coming from YouTube.

I noticed that the repo currently includes Build files (in `Debug\` and `tahm\Debug\`)
generally they are excluded from the rep. they take up unnecessary space, can cause conflicts between contributors, and aren't required for others to build the project themselves.

I suggest adding these to `.gitignore`:
```
Debug/
Release/
```
I saw that `pdb, obj, ...` files are included in `.gitignore`
so you may need to delete the files as well.

**Comment by tamtaasatiani** [comment]
debug folder will not be removed from the repository, because SDL dll files are inside the debug folder. if i do that, whoever wants to run the engine will no longer be able to run it right after cloning without adjustments

**Comment by alireza-keshavarz-d** [comment]
there are a few options that can be used for the build system, so more people can contribute (as of now only Visual Studio  (Windows) is supported).

`cmake` is the industry standard for building C/C++ project.
it can fetch SDL library automatically (so no worries there), it can be configured to build for other platforms and generate Solution files Visual Studio.

there are other options like `Ninja`, `Premake`, but i personally have not worked with them, so I prefer no to say anything about them

**Comment by horrified-dev** [comment]
I totally agree with the reply above.
use a build generator such as cmake or premake.
not only is it a good idea, but users from linux and macOS community will be able to use this and contribute to your engine further aswell.

**Comment by SirJamesClarkMaxwell** [comment]
Hi, I come from YT
And i fully agree with alumen and cherrypop. From my side i recommend the Premake. In my honest opinion it is so much easier to write and understand the code, with respect to the CMake.

**Comment by Lewi97** [comment]
I agree with all of the above, a build system will help with cross platform development and allow more people to work on the project. I would recommend using CMAKE and Vcpkg as dependency manager. A dependency manager like Vcpkg will make it so much easier to introduce new dependencies and avoid re-inventing the wheel.

**Comment by SirJamesClarkMaxwell** [comment]
@Lewi97  why VCPkg not premake?  AFAIK vcpkg install dependency globally on level system. I will not die for this sentence i used only once when i tried to integrate ImGui and I failed. 
If we are talking about restructurizing the project itself
i want to point three things, i mention this in yt comment also:
1. I wouldn't put dependencies code into `lib` directory. I would create a Vendor Directory and specify the output path for compiler where the files should be build (./lib) and (this is one line in premake) 
2. Second thing is that I probably would make a static library for Core engine which will be the base for other games. (yes i considered that this is not full game engine with editor etc.) 
3. I wrote this is different issue, but i quite don't get the architecuture of this engine. Why key parts are defined as nested classes inside one one big class and exposed as roll pointers in public section? (but this is discousion for another issue)

**Comment by Lewi97** [comment]
@SirJamesClarkMaxwell Vcpkg and premake are completely different software. Premake is a project files generator like CMAKE and Vcpkg is a dependency manager. 
With Vcpkg you no longer need to vendor third party code. Please read this guide on a very simple project using Vcpkg and CMAKE, its like magic how simple it makes dependency management: https://learn.microsoft.com/en-us/vcpkg/get_started/get-started .

**Comment by datuchela** [comment]
@tamtaasatiani, any thoughts on incorporating either of the multi-platform build systems mentioned above? It would be great if you or anyone else could do it. can’t really contribute from Linux without it.

**Comment by Javier-Lozano** [comment]
@datuchela I made a Makefile to compile it. I haven't done anything with tahmlib yet but it runs =P. You just need to install SDL2 development packages. I can contribute with that.

**Comment by Javier-Lozano** [comment]
@datuchela Copy this and save it to a file called `Makefile` put it in the same directory as `main.cpp` and trough a terminal just type `make` you'll need SDL2 and SDL2_ttf development packages installed. This will produce an executable named "test".
```
# Test Linux Makefile

CXX := g++
OPT := 
INC := `sdl2-config --cflags`
LIB := -lSDL2 -lSDL2_ttf
TAHM_SRC := $(wildcard tahm/*.cpp)
TAHM_OBJ := $(patsubst tahm/%.cpp,%.o,$(TAHM_SRC))
PROJECT = test

all: $(PROJECT)

debug: OPT = -g -Wall
debug: clean
debug: $(PROJECT)

$(PROJECT): $(TAHM_OBJ) main.cpp
	$(CXX) $(OPT) $(INC) main.cpp engine.cpp *.o -o $(PROJECT) $(LIB)

$(TAHM_OBJ): %.o: tahm/%.cpp
	$(CXX) $(OPT) -c $< $(INC) 

tell:
	@echo $(TAHM_OBJ)

.PHONY: clean
clean:
	rm -rf $(PROJECT)
	rm -rf *.o

```

**Comment by datuchela** [comment]
@Javier-Lozano Thanks for the help! works as expected.
